modules/autokey/iam.tf

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#Create KMS Service Agent
+resource "google_project_service_identity" "kms_service_agent" {
+  count    = local.create_autokey_key_handles ? 1 : 0
+  provider = google-beta
+
+  service = "cloudkms.googleapis.com"
+  project = data.google_project.kms_project.number
+}
+
+# Wait delay after creating service agent.
```

Comment:
Is there a deterministic way to avoid hard coding a wait?  For example, is this left over artifact from API enablement?

Reply:
This is a tricky point.
Theoretically, the API call and the Terraform resource apply waits for the SA agent to be created or the service to be enabled before continue;
In practice, we experienced a lot of propagation issues in the past where we receive the OK from API but when we try to use the resource, we face an error related to the service not being ready.

Some references from Foundation where we follow this same `time_sleep` approach for similar cases:
- [Ref 1](https://github.com/terraform-google-modules/terraform-example-foundation/blob/0bc707bc665d31ca9d20f8e984e780bf656d99e4/1-org/modules/cai-monitoring/main.tf#L60);
- [Ref 2](https://github.com/terraform-google-modules/terraform-example-foundation/blob/0bc707bc665d31ca9d20f8e984e780bf656d99e4/2-environments/modules/env_baseline/kms.tf#L31);
- [More](https://github.com/search?q=repo%3Aterraform-google-modules%2Fterraform-example-foundation%20time_sleep&type=code);

Reply:
Sg. Thanks for confirming